src/runtime/slice.go

```diff
@@ -47,7 +47,13 @@ func sliceGrow(oldBuf unsafe.Pointer, oldLen, oldCap, newCap, elemSize uintptr)
 	// memory allocators, this causes some difficult to debug issues.
 	newCap = 1 << bits.Len(uint(newCap))
 
-	buf := alloc(newCap*elemSize, nil)
+	var layout unsafe.Pointer
+	if elemSize == 1 {
```

Comment:
Could this check < size of pointer?

Reply:
Yeah, I was just being ultra-conservative for this initial POC, and also my test case was focussed on strings.  This could absolutely be smarter.